packages/cht/src/Adaptor.js

```diff
@@ -0,0 +1,129 @@
+import { expandReferences } from '@openfn/language-common/util';
+import * as util from './Utils';
+
+/**
+ * State object
+ * @typedef {Object} SHTHttpState
+ * @private
+ * @property data - the parsed response body
+ * @property response - the response from the SHT HTTP server, including headers, statusCode, body, etc
+ * @property references - an array of all previous data objects used in the Job
+ **/
+
+/**
+ * Options provided to the HTTP request
+ * @typedef {Object} RequestOptions
+ * @property {object|string} body - body data to append to the request. JSON will be converted to a string (but a content-type header will not be attached to the request).
+ * @property {object} errors - Map of errorCodes -> error messages, ie, `{ 404: 'Resource not found;' }`. Pass `false` to suppress errors for this code.
+ * @property {object} form - Pass a JSON object to be serialised into a multipart HTML form (as FormData) in the body.
+ * @property {object} query - An object of query parameters to be encoded into the URL.
+ * @property {object} headers - An object of headers to append to the request.
+ * @property {string} parseAs - Parse the response body as json, text or stream. By default will use the response headers.
+ * @property {number} timeout - Request timeout in ms. Default: 300 seconds.
+ */
+
+/**
+ * Make a GET request against the base URL.
+ * @example <caption>Get a list of contacts</caption>
+ * get("/api/v2/export/contacts");
+* @example <caption>Filter contacts given a name</caption>
+ * get("/api/v2/export/contacts", {
+  "filters": {
+    "search": "jim"
+  }
+});
+ * @function
+ * @public
+ * @param {string} path - Path to resource
+ * @param {RequestOptions} options - Optional to configure the HTTP request
+ * @param {function} [callback] - Optional callback to handle the response
+ * @returns {Operation}
+ * @state {SHTHttpState}
+ */
+export function get(path, options, callback) {
+  return request('GET', path, null, options, callback);
+}
+
+/**
+ * Make a POST request against the base url
+ * @example <caption>Create a new person</caption>
+ * post("/api/v1/people", {  
```

Comment:
I really like that you've taken these examples from the official docs. That's really smart!